client/components/migration/feature-announcement.jsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Flex, FlexItem, Modal, Icon, Button } from '@wordpress/components';
+import { useState, useEffect } from '@wordpress/element';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import bg from './images/wcshipping-migration.jpg';
+import { Dashboard, Preformatted, LessonPlan, Shipping } from './icons';
+import {
+	isEligableToMigrate,
+	wcshippingMigrationState,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors';
+import { installAndActivatePlugins } from './migration-runner';
+import { TIME_TO_REMMEMBER_DISMISSAL_SECONDS } from './constants';
+
+const FeatureAnnouncement = ( { translate, isEligable, previousMigrationState, onClose } ) => {
+	const [isOpen, setIsOpen] = useState(isEligable);
+	const [isUpdating, setIsUpdating] = useState(false);
+
+	useEffect( () => {
+		const isMigrationDismissed = window.wpCookies.get( 'wcst-wcshipping-migration-dismissed' ) && parseInt( window.wpCookies.get( 'wcst-wcshipping-migration-dismissed' ) );
+		if ( isMigrationDismissed ) {
+			setIsOpen( false );
+		}
+	}, [] );
+
+	const closeModal = () => {
+		setIsOpen(false);
+		onClose && onClose();
+	};
+
+	const snooze = () => {
+		window.wpCookies.set( 'wcst-wcshipping-migration-dismissed', 1, TIME_TO_REMMEMBER_DISMISSAL_SECONDS );
+		setIsOpen( false );
+		onClose && onClose();
+	};
+
+	const update = async () => {
+		setIsUpdating( true );
+		await installAndActivatePlugins( previousMigrationState );
+		setIsUpdating( false );
+	};
+
+	return <>{isOpen && (<Modal
+		className="migration__announcement-modal"
+		shouldCloseOnEsc={false}
+		shouldCloseOnClickOutside={false}
+		onRequestClose={closeModal}
+	>
+		<Flex gap={8}>
+			<FlexItem>
+				<Flex>
+					<span>
+						{translate('Update')}
+					</span>
+					<h2>
+						{translate('A new dedicated WooCommerce Shipping extension is now available')}
+					</h2>
+					<p>{translate('We\'ll ensure a smooth transition by providing the functionality to carry over all your settings and shipping labels when you update.')}</p>
```

Comment:
Very late/last minute suggestion: The "providing the functionality to carry" part made me go back and re-read it, so perhaps a simple change like this would be OK last minute? 😄 

```suggestion
					<p>{translate('We will ensure a seamless transition by transferring of all your settings and shipping labels when you update.')}</p>
```

Reply:
Looks good to me, i am committing your suggestion.

Reply:
Actually, I will change it to "We will ensure a seamless transition by transferring ~~of~~ all your settings and shipping labels when you update."

Reply:
Fixed in 71637e12d01a01cfeb932deebcfba170472d3e28